### ISSUE TITLE
Merge tykcommon repo with this one

### ISSUE THREAD

**Issue by mvdan** [TykTechnologies/tyk#390]
There is no advantage to keeping them as separate git repos. I see multiple disadvantages, such as:

* Separate issue trackers
* Changes affecting code in common like API definitions need two PRs
* Unnecessary vendoring

#389 should happen first.

**Comment by mvdan** [comment]
@lonelycode this is my proposal for the new package paths:

* `tykcommon.GetLogger()` -> `tyk/log.Get()`
* `tykcommon` API stuff -> `tyk/api/v0.*` (package name `api`)
* `tykcommon.coprocess.*` -> `tyk/coprocess.*

If it sounds OK with you, I'll handle the move.

**Comment by buger** [comment]
Just to clarify, this will be a new separate package in tyk repo, smth like `tyk/common`?

**Comment by mvdan** [comment]
No, when I say `tyk/log.Get` the package will be `tyk/log`. Same for the rest - they all hang from the root package here without a `common` in between. That's IMO as meaningless as `util`.

**Comment by lonelycode** [comment]
Wouldn't the entire `tyk` codebase be compiled into the binary for whatever is using it then? If I import a package from another large project, I thought the entire project doesn't get compiled in, just the package (it's already a big binary) - but I could be wrong?

Regarding the API Definition objects, I like the idea of versioning them, opens us up to change in the future, though when we were discussing versioning I was talking about the path interfaces exposed by the dashboard and the gateway, not the objects, so maybe `tyk/apidef/v0.*` would be better for the objects as it is different from `api`, which might end up being the actual service?

**Comment by mvdan** [comment]
I missed this comment, apologies.

> Wouldn't the entire tyk codebase be compiled into the binary for whatever is using it then? If I import a package from another large project, I thought the entire project doesn't get compiled in, just the package (it's already a big binary) - but I could be wrong?

If you import a Go package, only it and its dependencies will get linked. What I'm changing here is git repos - all the `tykcommon` stuff will keep on being separate Go packages. The new packages being subpackages of the big `tyk` one doesn't change anything. I don't undesrta

> Regarding the API Definition objects, I like the idea of versioning them, opens us up to change in the future, though when we were discussing versioning I was talking about the path interfaces exposed by the dashboard and the gateway, not the objects, so maybe tyk/apidef/v0.* would be better for the objects as it is different from api, which might end up being the actual service?

In general an `api` package means API objects. It's a name that you'll have to type very often, so IMO the shorter the better.

For the actual service you can come up with something like `tyk/gateway`, `tyk/service`, `tyk/runner`, etc.

**Comment by lonelycode** [comment]
> In general an api package means API objects. It's a name that you'll have to type very often, so IMO the shorter the better.

Still confusing: The Gateway API, API Definition objects for hosted proxies, API's have a dual meaning in Tyk, the one(s) being hosted and the actual C&C API. Would prefer this to be explicit in the naming, so `apid` is better than just `api`.

**Comment by mvdan** [comment]
To fix packages:

* `tykcommon-logger.GetLogger()` is now `tyk/log/Get()`
* `tykcommon.*` is now `tyk/apidef.*`
